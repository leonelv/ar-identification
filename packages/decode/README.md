
# @ar-identification/decode


A decoder for the Argentinian DNI PDF417


## Installation


```bash
yarn add @ar-identification/decode@0.1.0
```
    
## Usage/Examples

The library includes a function called `dniDecoder` which decodes the data from the PDF417 code.

### The DNI Object

| prop  | type signature  | description  |
|---|---|---|
| `name` |  `string` |  Nombre / Name |
| `surname`  |  `string` |  Apellido / Surname |
| `dateOfBirth`  | `Date`  | Fecha de nacimiento / Date of birth  |
| `dateOfIssue`  | `Date`  | Fecha de emisión / Date of issue  |
| `dateOfExpiry`  | `Date`  | Fecha de vencimiento / Date of expiry |
| `copy`  | `string`  | Ejemplar |
| `sex`  | `"MALE" \| "FEMALE" \| "NON_BINARY"`  | Sexo / Sex  |
| `dni`  | `string`  | Documento / Document |
| `cuil`  | `string`  | CUIL |
| `id`  | `string`  | Tramite N° / Of. Ident |


### Implementation
```tsx
import { dniDecoder } from "@ar-identification/decode";

const dni = await decode("00123456789@PITO@ELSA@F@1234567@B@22/11/1998@12/03/2018@204"); //returns a DNI object
```

