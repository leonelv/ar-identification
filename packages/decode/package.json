{
  "name": "@ar-identification/decode",
  "description": "Argentinian identification data decoder",
  "source": "./src/index.ts",
  "module": "./dist/decode.js",
  "types": "./dist/index.d.ts",
  "version": "0.1.0",
  "scripts": {
    "build": "microbundle -f esm",
    "dev": "yarn build --watch"
  },
  "dependencies": {
    "dayjs": "^1.11.3",
    "zod": "^3.17.3"
  }
}
