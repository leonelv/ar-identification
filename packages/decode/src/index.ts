export { default as dniDecoder } from "./dni-decoder";
export * from "./types";
