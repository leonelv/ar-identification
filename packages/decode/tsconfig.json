{
  "compilerOptions": {
    "module": "ESNext",
    "target": "ESNext",
    "allowSyntheticDefaultImports": true,
    "moduleResolution": "node",
    "declaration": true,
    "declarationDir": "./dist"
  },
  "include": ["src/**/*"],
  "exclude": ["src/**/*.spec.ts"]
}
