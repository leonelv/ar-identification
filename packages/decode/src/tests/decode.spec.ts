import decode from "../dni-decoder";

describe("decode", () => {
  test("should return a parsed DNI object", async () => {
    const dni = await decode("00123456789@PITO@ELSA@F@1234567@B@22/11/1998@12/03/2018@204");

    expect(dni).toMatchSnapshot()
  });
});
