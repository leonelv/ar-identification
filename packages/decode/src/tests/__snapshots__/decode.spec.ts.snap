// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode should return a parsed DNI object 1`] = `
Object {
  "copy": "B",
  "cuil": "20-1234567-4",
  "dateOfBirth": 1998-11-22T03:00:00.000Z,
  "dateOfExpiry": 2033-03-12T03:00:00.000Z,
  "dateOfIssue": 2018-03-12T03:00:00.000Z,
  "dni": "1234567",
  "id": "00123456789",
  "name": "ELSA",
  "sex": "FEMALE",
  "surname": "PITO",
}
`;
